src/node/collapser.ml

```diff
@@ -120,13 +128,19 @@ let collapse_until (type s) (tlog_coll:Tlogcollection.tlog_collection)
                     begin
                       maybe_log pi >>= fun () ->
                       begin
-                        if Sn.rem pi tlog_entries_per_file = 0L
+                        if pi = !next_rollover_point
                         then
-
-                          cb (Sn.of_int (first_tlog + !processed)) >>= fun () ->
-                          Lwt.return( processed := !processed + 1 )
+                          begin
+                            Logger.info_f_ "rollover_point pi:%Li next_rollover_point =%s"
+                                          pi (Sn.string_of !next_rollover_point)
+                            >>= fun () ->
+                            cb (first_tlog + !processed) >>= fun () ->
+                            let () = incr processed in
+                            let () = next_rollover_point := get_next_point (Sn.succ pi) in
```

Comment:
if you have: I as last entry of tlog_{n}, and I also as first entry in tlog_{n+1}, then the tlogmap will start folding at tlog_{n + 1} when starting at I, which is ok.